DataFormats/ParticleFlowReco/interface/PFRecHit.h

```diff
@@ -53,11 +54,10 @@ namespace reco {
     /// default constructor. Sets energy and position to zero
     PFRecHit(){}
 
-    PFRecHit(CaloCellGeometry const * caloCell, unsigned int detId,
-             PFLayer::Layer layer,
-             float energy) :
-        caloCell_(caloCell),  detId_(detId),
-        layer_(layer), energy_(energy){}
+  PFRecHit(std::shared_ptr<const CaloCellGeometry> caloCell, 
+	   unsigned int detId, PFLayer::Layer layer,
+	   float energy) : caloCell_(std::move(caloCell)),  detId_(detId),
```

Comment:
This was recommended by Chris

Reply:
@ianna The original caloCell is a copy passed to the function. No other code has access to the copy and therefore it is safe to modify. This is now a new C++ 'idiom' recommended by C++ experts.

Reply:
@Dr15Jones - is it the original shared pointer without destructor?

Reply:
So in
```C++
std::shared_ptr<const CaloCellGeometry> cell = ...; //get the cell from the geometry
//shared_ptr ref count is now  1 

PFRecHit hit( cell, ... );
//shared_ptr ref count is now  2
```
During the call to PFRecHit constructor, cell gets copied (since the argument is passed by value) then the copied gets moved to the newly constructed member data PFRecHit::caloCell_. During the move the pointer and the ref count are moved to caloCell_ while the original argument gets the pointer and the ref count removed (but without decrementing the ref count). Therefore when the argument goes out of scope, it has nothing to delete.